pinot-spi/src/main/java/org/apache/pinot/spi/stream/PartitionGroupInfo.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.stream;
+
+/**
+ * A PartitionGroup is a group of partitions/shards that the same consumer should consume from.
```

Comment:
Do we need this class? It seems to be used as a List<PartitionGroupInfo> all the time. We can just use a  Map<Integer, StrramPartitionMsgOffset> instead. 
Or, do you plan to add more members to the class later?

Reply:
I feel the method is more readable by adding a concrete class there. Especially, the getPartitionGroupInfoList, which now reads as "give current partitionGroups as List<PartitionGroupMetadata and get back new partitionGroups as List<PartitionGroupInfo"